leftwm-core/src/utils/child_process.rs

```diff
@@ -87,6 +88,31 @@ impl Nanny {
     /// Could be caused by inadequate permissions.
     pub fn run_global_up_script() -> Result<Option<Child>> {
         let mut path = Self::get_config_dir()?;
+        let config_dir = fs::read_dir(&path)?;
+
+        let mut scripts = std::collections::BinaryHeap::new();
+
+        for entry in config_dir {
+            let entry = match entry {
+                Ok(e) => e,
+                Err(_) => continue,
+            };
+            let file = entry.path();
+
+            if let Some(extension) = file.extension() {
+                if extension == "up" {
+                    scripts.push(Reverse(file));
+                }
+            }
+        }
```

Comment:
this code-part has its own logic, so I'd move it out to a function.

Also does your scripts-collection needs to be a Binary-Heap? If I'm seeing it right, you're just collecting the values, this can be done with a `Vec` which has (amortized) O(1) whichis definetely better than O(log(n)).

Reply:
> this code-part has its own logic, so I'd move it out to a function.
I agree.

As for the `BinaryHeap` vs `Vec` I choose the former because we want (as discussed on discord) to have an order in the executing of the scripts so if using a `Vec` we should `.sort()` it before executing the scripts.
To be fair most users won't have more than an handful of scripts so it doesn't really matter. I've chosen the BinaryHeap because of the emitted asm code size.
Here is the link to the example: https://godbolt.org/z/sMqocbEhM

What do you think?

Reply:
> As for the BinaryHeap vs Vec I choose the former because we want (as discussed on discord) to have an order in the executing of the scripts so if using a Vec we should .sort() it before executing the scripts.
To be fair most users won't have more than an handful of scripts so it doesn't really matter. I've chosen the BinaryHeap because of the emitted asm code size.
Here is the link to the example: https://godbolt.org/z/sMqocbEhM
> 
> What do you think? 

Oh ok, I didn't knew that you discussed that on discord. Your argumentation sounds good to me.